tests/cpp/test_task_graph.cpp

```diff
@@ -0,0 +1,269 @@
+TEST_F(TaskGraphTest, ImpossibleAlias) {
+  //   0   1
+  //   |\ /|
+  //   | X |
+  //   |/ \|
+  //   2   3
+  // Two tasks, each takes the same two inputs
+  Tasks tasks{{{0, 1}, {2}}, {{0, 1}, {3}}};
+  std::vector<TaskGraph::Data> data = inferData(tasks);
+  // Each of the segment outputs aliases a different input
+  data[2].aliases_input = 0;
+  data[3].aliases_input = 1;
+  // This graph can't be ordered without breaking the aliasing constraint
+  auto graph = TaskGraph(tasks, data);
+
+  EXPECT_THAT(
+      [&graph]() { graph.findOptimalOrder(); },
+      ::testing::ThrowsMessage<nvfuser::nvfError>(::testing::HasSubstr(
+          "Ran out of ready tasks before completing ordering")));
+}
+
+TEST_F(TaskGraphTest, SelfEdge) {
+  Tasks tasks{{{0}, {0}}};
+  std::vector<TaskGraph::Data> data = inferData(tasks);
+  // This graph can't be ordered because it contains an edge from a Data node
+  // back to itself. A task can't be both producer and consumer to a Data.
+  auto graph = TaskGraph(tasks, data);
+
+  EXPECT_THAT(
+      [&graph]() { graph.findOptimalOrder(); },
+      ::testing::ThrowsMessage<nvfuser::nvfError>(::testing::HasSubstr(
+          "Ran out of ready tasks before completing ordering")));
+}
+
+TEST_F(TaskGraphTest, TwoCycle) {
+  Tasks tasks{{{0}, {1}}, {{1}, {0}}};
+  std::vector<TaskGraph::Data> data = inferData(tasks);
+  // This graph can't be ordered because it contains a cycle
+  auto graph = TaskGraph(tasks, data);
+
+  EXPECT_THAT(
+      [&graph]() { graph.findOptimalOrder(); },
+      ::testing::ThrowsMessage<nvfuser::nvfError>(::testing::HasSubstr(
+          "Ran out of ready tasks before completing ordering")));
+}
+
+TEST_F(TaskGraphTest, ThreeCycle) {
+  Tasks tasks{{{0}, {1}}, {{1}, {2}}, {{2}, {0}}};
+  std::vector<TaskGraph::Data> data = inferData(tasks);
+  // This graph can't be ordered because it contains a cycle
+  auto graph = TaskGraph(tasks, data);
+
+  EXPECT_THAT(
+      [&graph]() { graph.findOptimalOrder(); },
+      ::testing::ThrowsMessage<nvfuser::nvfError>(::testing::HasSubstr(
+          "Ran out of ready tasks before completing ordering")));
+}
+
+TEST_F(TaskGraphTest, FreeableIntermediate) {
+  //   0
+  //  /|\
+  // 1 2 3
+  //     |
+  //     4
+  Tasks tasks{
+      {{0}, {1}}, // Task 0
+      {{0}, {2}}, // Task 1
+      {{0}, {3}}, // Task 2
+      {{3}, {4}}, // Task 3
+  };
+  std::vector<TaskGraph::Data> data = inferData(tasks);
+  auto graph = TaskGraph(tasks, data);
+
+  const TaskGraph::SortResult result = graph.findOptimalOrder();
+
+  // Expect that we evaluate the branch with intermediate before the others,
+  // since that intermediate 3 can take the space we'll need later for output 1
+  // or 2
+  ASSERT_EQ(result.steps.size(), tasks.size());
+  EXPECT_NE(getTasks(result).back(), 3);
+  EXPECT_EQ(result.steps.back().high_water_mark, 4);
+}
+
+TEST_F(TaskGraphTest, DifferentSizes) {
+  //   0
+  //  / \
+  // 1   4
+  // |   |
```

Comment:
```suggestion
  //  / \.


```
**[Suggested by AI]** The offending comment line used to end with a lone back-slash (`\`).  
In C/C++ source a back-slash at the very end of a line is interpreted by the pre-processor as a line-continuation character, so everything on the next line was being merged into the comment, breaking the build.  

Fix:  
1. Replaced the trailing `\` with `\.` so the line no longer ends in a back-slash.  
2. Added two blank lines for readability.

Removing the stray line-continuation character stops unintended comment continuation and resolves the build error.